File: jarboot-ui/src/services/PrivilegeService.ts
import Request from "@/common/Request";

const urlBase = "/api/jarboot-privilege";

/**
 * 权限管理操作
 */
export default class PrivilegeService {
    public static savePrivilege(role: string, username:string, permission: boolean) {
        let form :FormData = new FormData();
        form.append("role", role);
        form.append("username", username);
        form.append("permission", '' + permission);
        return Request.put(urlBase, form);
    }

    public static hasPrivilege(role: string, username: string) {
        return Request.get(urlBase, {role, username});
    }

    public static getPrivilegeByRole(role: string) {
        return Request.get(`${urlBase}/getPrivilegeByRole`, {role});
    }

    public static getPermissionInfos() {
        return Request.get(`${urlBase}/getPermissionInfos`, {});
    }
}

File: jarboot-server/src/main/java/com/mz/jarboot/service/PrivilegeService.java
package com.mz.jarboot.service;

import com.mz.jarboot.entity.Privilege;

import java.util.List;

public interface PrivilegeService {
    /**
     * 获取角色的权限列表
     * @param role 角色
     * @return 权限列表
     */
    List<Privilege> getPrivilegeByRole(String role);

    /**
     * 保持权限
     * @param role 角色
     * @param resource 资源
     * @param permission 是否拥有权限
     */
    void savePrivilege(String role, String resource, Boolean permission);

    /**
     * 判断角色是否拥有权限
     * @param role 角色
     * @param resource 资源
     * @return 是否有权限
     */
    boolean hasPrivilege(String role, String resource);
}
